bool stop_request
---
bool stop_result